src/firewall/core/rich.py

```diff
@@ -778,6 +827,10 @@ def _import_from_string(self, rule_str):
 
             index = index + 1
 
+    def _check_entry(self, entry):
+        if entry is not None:
+            entry.check(family=self.family)
+
```

Comment:
IMO it's not useful to abstract the if statement for a "null check". Just do it at the call sites.

e.g.
```python
if self.source is not None:
  self.source.check()
```

Reply:
I'd rather not.

Your example is incorrect. The full replacement also requires to pass the family, so it's
```python
if self.source is not None:
    self.source.check(family=self.family)
```
which then needs to be copied at 6 places, and adjusted for `self.{source,destination,element,log,audit,action}`.

That redundancy and verbosity is avoided by `_check_entry()`.

Reply:
> Your example is incorrect.

Sure. I forgot passing `family` to `check()` because above I suggested removing it entirely. The point I was making is still stands.

> which then needs to be copied at 6 places, and adjusted for

It would only be copying the `if foo is not None:`. I don't care enough about this to discuss further. What you have is okay.